arrays/arrays.c

```diff
@@ -99,15 +99,25 @@ char *arr_read(Array *arr, int index) {
 void arr_insert(Array *arr, char *element, int index) {
 
   // Throw an error if the index is greater than the current count
-
+  if(index >= arr->count){
+    printf("Array capacity has been reached");
+    return 0;
+  }
   // Resize the array if the number of elements is over capacity
-
+  if(arr->count+1 > arr->capacity){
+    resize_array(arr);
+  }
+  
   // Move every element after the insert index to the right one position
-
+  for(int i = index +1; i<arr->count; i++){
+    arr->elements[i+1] = arr->elements[i];
+  }
   // Copy the element and add it to the array
+  char *element_cpy = element;
```

Comment:
As it is now, the element_cpy pointer will point to the string pointed to by the input element pointer. What you want is to malloc a chunk of memory, write the string to the chunk of memory, and have the element_cpy pointer point to this newly copied string. The strdup() function does this for us. Brady went into this a little bit during lecture.